src/python/gudhi/periodic_cubical_complex.pyx

```diff
@@ -77,34 +86,34 @@ cdef class PeriodicCubicalComplex:
         """
 
     # The real cython constructor
-    def __cinit__(self, dimensions=None, top_dimensional_cells=None,
+    def __cinit__(self, dimensions=None, cells=None, input_top_cells=True,
```

Comment:
Renaming the argument breaks backward compatibility in Python. It is possible, but I would prefer to explore alternatives first. Could we imagine keeping `top_dimensional_cells`, adding a parameter `vertices`, and giving an error if both (or none) are specified?

You may want to take a look at the 2 branches proposing a new interface for cubical complexes (sklearn & tensorflow) to check if they will be able to accommodate this new option easily.